README.md

```diff
@@ -492,6 +495,92 @@ status:
+3. The `env` field in the application resource **MUST** conform to the [EnvVar][env] API.
+4. The `containers` field in the application resource **MUST** conform to the API schema given here:
+
+    ```yaml
+    schema:
+      openAPIV3Schema:
+        properties:
+          containers:
+            description: Containers describes which containers in a Pod should be bound to
+            items:
+              anyOf:
+              - type: integer
+              - type: string
+              x-kubernetes-int-or-string: true
+            type: array
+    ```
+
+A `CustomBindingPath` resource **MUST** define a `.status.conditions` which is an array of `Condition` objects.  A `Condition` object **MUST** define `type`, `status`, and `lastTransitionTime` entries.  At least one condition containing a `type` of `Ready` **MUST** be defined.  The `status` of the `Ready` condition **MUST** have a value of `True`, `False`, or `Unknown`.  The `lastTransitionTime` **MUST** contain the last time that the condition transitioned from one status to another.  A `CustomBindingPath` resource **MAY** define `reason` and `message` entries to describe the last `status` transition.  There could be multiple application resources matching the `apiVersion` and `kind`.  Partial failures **MUST** be aggregated and reported on the binding status's `Ready` condition.
+
+The reconciler **MUST** set the `CustomBindingPath`'s `Ready` condition according to the same rules set in [Ready Condition Status](#ready-condition-status) for `ServiceBinding` resource.
+
+[cbp-crd]: internal.service.binding_bindingpathmaps.yaml
+[volume]: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#volume-v1-core
+[volumemount]: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#volumemount-v1-core
+[env]: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#envvar-v1-core
+
+### Resource Type Schema
+
+```yaml
+apiVersion: internal.service.binding/v1alpha2
+kind: CustomBindingPath
+metadata:
+  name:                 # string
+  generation:           # int64, defined by the Kubernetes control plane
+  ...
+spec:
+  application:          # ObjectReference-like
+    apiVersion:         # string
+    kind:               # string
+
+  bindingPaths:
+    volume:             # string, optional
+    volumeMount:        # string, optional, volume must be specified along with this
+    env:                # string, optional
+    containers:         # string, optional
+
+status:
+  conditions:           # []Condition containing at least one entry for `Ready`
+  - type:               # string
+    status:             # string
+    lastTransitionTime: # Time
+    reason:             # string
+    message:            # string
+  observedGeneration:   # int64
+```
+
+### Custom Binding Paths Example
+
+```yaml
+apiVersion: internal.service.binding/v1alpha2
+kind: BindingPathMap
+metadata:
+  name: example-binding-paths
```

Comment:
Can we make the name deterministic, similar to the name for CRDs and API Services, matching `{version}.{resource}.{apiGroup}`?

For this example, that would be something like:

```suggestion
  name: v1alpha1.exampleappbuilders.apps.examples.org
```

Reply:
Done. 56f767de1c436325555863434e297ad7da9b8fd5